src/nvim/ops.c

```diff
@@ -3970,7 +3972,9 @@ int do_join(size_t count, int insert_space, int save_undo, int use_formatoptions
   int remove_comments = (use_formatoptions == true)
                         && has_format_option(FO_REMOVE_COMS);
   bool prev_was_comment = false;
-  assert(count >= 1);
+  // don't send change if line is truncated before
+  bool can_send_change = ((curbuf->b_ml.ml_flags & ML_LINE_DIRTY) != 0)
```

Comment:
Not sure if the conditions here are sufficient or correct. Maybe add a pending operation check?